src/win/fs.c

```diff
@@ -148,6 +148,72 @@ void uv_fs_init(void) {
 }
 
 
+static int32_t fs__decode_wtf8_char(const char** input) {
+  uint32_t code_point;
+  uint8_t b, b2, b3, b4;
+
+  b = **input;
+  if (b <= 0x7F) {
+    code_point = b;
+  } else if (b >= 0xC2 && b <= 0xDF) {
+    if (!(*input)[1])
+      return -1;
+    b2 = *++*input;
```

Comment:
nit: I think it's okay if the iterator is left in an invalid state if this test fails, so that we can write it more simply as:
```suggestion
    b2 = *++*input;
    if (b2 == '\0' || (b2 & 0xc0) != 0x80)
      return -1;
```
(and so on)